core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -891,7 +891,11 @@ public TimeValue ackTimeout() {
 
         @Override
         public void run() {
-            runTasksForExecutor(executor);
+            // if this task is already processed, it shouldn't trigger another executing of pending
+            // tasks for the same executor, to be fair to other executors
```

Comment:
maybe more like this:
if this task is already processed, the executor shouldn't execute other tasks (that arrived later), to give other executors a chance to execute their tasks.